code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/RegionMonitor.java

```diff
@@ -154,14 +161,11 @@ private void cleanLinks (List<AccountLink> accountLinks, Map<String, ExternalReg
                         if(accLink == null || removedStates.contains(accLink.getState())){
                             objectProcessManager.executeStandardProcess(StandardProcess.REMOVE, link, null);
                         }
-                } else {
-                    existingLinks.add(externalProjectKey);
                 }
+                existingLinks.add(externalProjectKey);
```

Comment:
we should skip this when the link gets cleaned up

Reply:
added continue above